### ISSUE TITLE
Confusing errors generated with named operator methods

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#17853]
It seems that the compiler currently does not prevent the user from creating arbitrarily named `operator`s, and this can result in confusion if someone (like me, ahem) accidentally types `operator` where they meant to type `proc`.  For example, this afternoon, I wrote something like:

```chapel
record R {
  operator foo() {
    writeln("In operator foo()");
  }
}

var myR: R;

myR.foo();
```

and got the confusing error:

```
namedOperator2.chpl:9: error: unresolved call 'R.foo()'
namedOperator2.chpl:2: note: this candidate did not match: R.foo()
namedOperator2.chpl:9: note: because call includes 0 arguments
namedOperator2.chpl:2: note: but function can only accept 0 arguments
```

**Comment by bradcray** [comment]
It seems as though one way to solve this would be to refactor the `fn_ident` rule in the parser into an `op_ident` case that had all of the operators and to only permit `op_ident` to follow `operator` keywords (though this would result in a mere "syntax error" rather than something saying "You can't name an operator 'foo'", say).  But it would save the hassle of having to enumerate all the operators within `build.cpp` to make sure that operators only had legal names.  Or, maybe it would be sufficient to check that the operator didn't start with an alphanumeric character or `_` or `$` or whatever other non-operator characters we permit in identifiers?

**Comment by mppf** [comment]
AFAIK we already have some functions that enumerate all the operator names. @lydia-duncan  could probably quickly tell us the name of such a function.

**Comment by lydia-duncan** [comment]
Yeah, you want `isAstrOpName`.  I'd add such a functionality to checkParsed.cpp's `checkOperator` function which already handles the deprecation warning when not using the operator keyword.

**Comment by lydia-duncan** [comment]
This should be almost trivial to implement

**Comment by bradcray** [comment]
Just to make sure nobody's reading too much between the lines:  I got bitten by this, so filed the issue, but am not planning on fixing it.

**Comment by lydia-duncan** [comment]
Would you like me to put it in my sprint?

**Comment by bradcray** [comment]
I think it would be good to resolve before the next release, but it doesn't really matter to me whether it happens now or later.

**Comment by bradcray** [comment]
(and since we were discussing https://github.com/Cray/chapel-private/issues/1927, adding it to that list might be good as well—where I might put it after "Deprecate use M except *" in the smaller efforts list).
